src/modules/0.0.6/aws_ec2/index.ts

```diff
@@ -125,5 +135,65 @@ export const AwsEc2Module: Module2 = new Module2({
         },
       }),
     }),
+    registeredInstance: new Mapper2<RegisteredInstance>({
+      entity: RegisteredInstance,
+      entityId: (e: RegisteredInstance) => `${e.instance.id}|${e.targetGroup.targetGroupName}` ?? '',
+      equals: (a: RegisteredInstance, b: RegisteredInstance) => Object.is(a.port, b.port),
+      source: 'db',
+      cloud: new Crud2({
+        create: async (es: RegisteredInstance[], ctx: Context) => {
+          const client = await ctx.getAwsClient() as AWS;
+          const out = [];
+          for (const e of es) {
+            if (!e.instance?.instanceId || !e.targetGroup?.targetGroupArn) throw new Error('Valid targetGroup and instance needed.');
+            await client.registerInstance(e.instance.instanceId, e.targetGroup.targetGroupArn, e.port);
+            const registeredInstances = await AwsEc2Module.mappers.registeredInstance.cloud.read(ctx);
+            const relevantRegister = registeredInstances.find((ri: RegisteredInstance) => Object.is(ri.instance.instanceId, e.instance.instanceId) &&
+              Object.is(ri.targetGroup.targetGroupName, e.targetGroup.targetGroupName));
+            await AwsEc2Module.mappers.registeredInstance.db.update(relevantRegister, ctx);
+            out.push(relevantRegister);
+          }
+          return out;
+        },
+        read: async (ctx: Context, id?: string) => {
+          const client = await ctx.getAwsClient() as AWS;
+          if (id) {
+            // Just return empty array. Since theres no really a cloud id this should never be called.
```

Comment:
The faster apply change we want to do would fail on this logic. Why not just put the ID check at the end and filter to the relevant record instead? Same LOC and the function does what it should.

Reply:
Hmm because there's no really a unique I'd. It is a composite key in the database and the entityId custom function is a concatenation of id and target_group_name